### DISCUSSION TITLE
Could we streamline the metadata section of our ADVS peds template?

### DISCUSSION THREAD

**Discussion by rossfarrugia** [pharmaverse/admiralpeds#94] [General]
When implementing this for Roche, we got some feedback to consider:

1. We store in admiralpeds the raw metadata (growth charts) from WHO/CDC and then we manipulate them to our expected standardised format - so could we not just include the standardised datasets in our package too? So if a company is happy to follow exactly as we have then they wouldn't need to repeatedly run all the metadata lines of our template to re-generate the same metadata each time?
2. Or could the metadata processing be wrapped into a function as its quite repetitive if we expect most companies won't need to change? An argument could be included for age unit (days/weeks/months) as dependent on this the metadata processing changes and you'd need different metadata datasets generated, e.g. for months you wouldn't need to interpolate for CDC.

I explained the rationale to have it transparently laid out in the template is that users could actually choose to replace the CDC and WHO files with any growth charts they like (e.g. there's also some given by International Obesity TaskForce) so we wanted to show openly how we structure these files so that they could equally easily create their own if they ever needed. Although maybe if we had it in a function or provided dataset instead though its not so bad, as we could instead just explain it all in documentation.

**Comment by rossfarrugia** [comment]
David ideas shared from email which we can discuss at team meeting:

 I agree with the first discussion point: user coding should be streamlined. 

My cdcanthro function requires just one line of code, such as
cdcanthro(data, age_in_months, weight_kg, height_cm, bmi) (https://github.com/CDC-DNPAO/CDCAnthro)

WHO's 'anthro' function on CRAN works similarly (1 line for the user).

Maybe the long data set could be made into a wide dataset, ALL calculations (weight, height, BMI, etc) performed, concerverted back to long, and then merged/joined with the original data. (There are very fast pivot and join functions in the 'collapse' packages; these are much faster than the dplyr equivalents.

**Comment by Fanny-Gautier** [comment]
The current way of having the metadata code included in the template was chosen by the {admiralpeds} team for transparency reasons : 
-  when submitted, it is clear for the FDA how the various metadata are created (from WHO and CDC, derived in `DAYS` unit)
- users might want to update the metadata with different units or variables or even completely different metadata: it is thus much easier for them to directly see the code and the required the format to update as necessary
- if not needed by the user, this part of code can simply be dropped from the copied template

It will currently be left as it is until further feedback.
